src/net/connection/connection_pool.rs

```diff
@@ -88,13 +96,16 @@ impl ConnectionPool {
 
     /// Check if there are any connections that have not been active for the given Duration.
     fn check_for_timeouts(
-        connections: &Connections,
+        connections: &mut Connections,
```

Comment:
This should accept a `Arc<RwLock<_>>` and as said above only read when checking, and only getting the write lock on the map if we need to remove.